src/commit.rs

```diff
@@ -183,6 +183,20 @@ impl<'repo> Commit<'repo> {
         }
     }
 
+    /// Get the author of this commit, using the mailmap to map names and email
+    /// addresses to canonical real names and email addresses.
+    pub fn author_with_mailmap(&self, mailmap: &Mailmap<'_>) -> Result<Signature<'static>, Error> {
```

Comment:
`Signature<'static>` because `git_commit_author_with_mailmap` (`git_commit_committer_with_mailmap`) allocates a new signature the caller must free. `Signature::new` has the same return type.

Reply:
Yeah that should be fine I think!